geonode/layers/views.py

```diff
@@ -653,6 +654,58 @@ def layer_thumbnail(request, layername):
             )
 
 
+def layer_list(request):
+    """Get Layer list as JSON
+    """
+    if request.method == 'GET':
+        if 'q' in request.GET:
+            query = request.GET.get('q')
+
+            layers = Layer.objects.filter(
+                name__istartswith=query)
+            result = []
+            for layer in layers:
+                result.append(layer.name)
+        else:
+            layers = Layer.objects.all()
+            result = []
+            for layer in layers:
+                result.append(layer.name)
+        result = json.dumps(result)
+        return HttpResponse(result, content_type='application/json')
+
+
+def get_layer(request, layername):
```

Comment:
Hmm, I never know that there is API for it. Unfortunately, our current view, using layername (almost everywhere). We can move to use the API later (if it's ok to not do it now).